### ISSUE TITLE
[BUG] Annotation with data providers

### ISSUE THREAD

**Issue by MGatner** [NexusPHP/tachycardia#7]
**Environment:**
* PHP version: 7.4
* Tachycardia version: `v1.1.1`

**Description:**
When using the `@timeLimit` annotation on a test with a data provider the new limit is not respected.

**Steps to Reproduce:**
Here's my actual test case (using `ControllerTester`):
```
	/**
	 * @dataProvider indexDataProvider
	 * @timeLimit 0.75
	 */
	public function testIndexHasCorrectJobs(string $method, array $expected)
	{
		$result = $this->execute($method);
		$result->assertOK();
		$result->assertStatus(200);

		// Check that each expected Job is present and others are not
		foreach (['staff', 'client', 'completed', 'deleted'] as $type)
		{
			if (in_array($type, $expected))
			{
				// Check for the Job name linked by its ID
				$result->see(anchor('jobs/show/' . $this->$type->id, $this->$type->name));
			}
			else
			{
				$result->dontSee($this->$type->name);
			}
		}
	}

	public function indexDataProvider(): array
	{
		return [
			['staff', ['staff']],
			['active', ['staff', 'client']],
			['archive', ['completed']],
			['all', ['staff', 'client', 'completed']],
			['trash', ['deleted']],
		];
	}
```
 No matter what I set for `@timeLimit` I always get a warning:
```
Nexus\PHPUnit\Extension\Tachycardia identified this sole slow test:
+--------------------------------------------------------------+---------------+-------------+
| Test Case                                                    | Time Consumed | Time Limit  |
+--------------------------------------------------------------+---------------+-------------+
| JobsControllerTest::testIndexHasCorrectJobs with data set #2 | 00:00:00.68   | 00:00:00.50 |
+--------------------------------------------------------------+---------------+-------------+
```
Notice that "Time Limit" is `0.5`.

**Expected Output:**
Each iteration of a data-provided test should use the limit override.

**Comment by paulbalandan** [comment]
I have a hunch on this. I think the problem is in the `getAnnotations` method. I'm on mobile so probably will get back to PC by tomorrow.

Can you try this debug script?
```php
use PHPUnit\Unit\Test as TestUtil;

$annotations = TestUtil::parseTestMethodAnnotations(JobsControllerTest::class, 'testIndexHasCorrectJobs with data set #2');

var_dump($annotations);
// this should have the timeLimit subkey for the method key
```

I'm suspecting `TestUtil` needs to parse only the test method name without the dataname.

**Comment by MGatner** [comment]
```
┌────────────────────────────────────────────────────────────────────────────────────────┐
│ $annotations
└────────────────────────────────────────────────────────────────────────────────────────┘
array (2) [
    'method' => null
    'class' => array (1) [
        'mixin' => array (1) [
            0 => string (14) "CIUnitTestCase"
        ]
    ]
]
```

**Comment by paulbalandan** [comment]
Fixed in [f9750f6](https://github.com/NexusPHP/tachycardia/commit/f9750f6fac13213649a72f90e58f2e28d9b1ac6d).

**Comment by MGatner** [comment]
Confirmed, override time limit now shows:
```
Nexus\PHPUnit\Extension\Tachycardia identified this sole slow test:
+--------------------------------------------------------------+---------------+-------------+
| Test Case                                                    | Time Consumed | Time Limit  |
+--------------------------------------------------------------+---------------+-------------+
| JobsControllerTest::testIndexHasCorrectJobs with data set #2 | 00:00:01.06   | 00:00:00.75 |
+--------------------------------------------------------------+---------------+-------------+
```
